boards/nucleo-f042k6/include/periph_conf.h

```diff
@@ -19,12 +19,13 @@
 #ifndef PERIPH_CONF_H
 #define PERIPH_CONF_H
 
-/* No HSE available on this board */
-#define CLOCK_HSE            (0U)
-
 /* Adjust PLL factors when PLL is clocked by HSI */
-#define CLOCK_PLL_PREDIV     (2)
-#define CLOCK_PLL_MUL        (12)
+#ifndef CONFIG_CLOCK_PLL_PREDIV
+#define CONFIG_CLOCK_PLL_PREDIV     (2)
+#endif
+#ifndef CONFIG_CLOCK_PLL_MUL
+#define CONFIG_CLOCK_PLL_MUL        (12)
+#endif
```

Comment:
I know this comes from a previous PR but it could be usefull to have here a notion on how these values impact the PLL clock.

Reply:
Or somewhere BTW since this information seems to have been lost.

Reply:
It's not lost, it's an addition of this PR, available here: https://github.com/RIOT-OS/RIOT/pull/14921/files#diff-ef231d82dd626d4b69752305fbc8f46cR111

Reply:
Hmm this is not what I'm taking about, I'm looking for something like https://github.com/RIOT-OS/RIOT/blob/16271f253bd7e9f4efde6dfd23a7d8929e34de40/boards/p-nucleo-wb55/include/periph_conf.h#L62

Reply:
This is not that far from
```c
#define CLOCK_CORECLOCK                 ((CLOCK_PLL_SRC / CONFIG_CLOCK_PLL_PREDIV) * CONFIG_CLOCK_PLL_MUL)
```

but I can add a comment above the CLOCK_CORECLOCK define.

Reply:
I was thinking that its now a bit hidden what the current `CLOCK_CORECLOCK` is,  so a comment with what the default values is could be useful, so some version of:

```
/* CORECLOCK default 64MHz */
#define CLOCK_CORECLOCK                 ((CLOCK_PLL_SRC / CONFIG_CLOCK_PLL_PREDIV) * CONFIG_CLOCK_PLL_MUL)
```

But if custom values are used, for whatever reason also have that info in `periph_conf.h`:

```
/* CORECLOCK default 64MHz, CORECLOCK = (((PLL_IN / M) * N) / R) */
#ifndef CONFIG_CLOCK_PLL_PREDIV
#define CONFIG_CLOCK_PLL_PREDIV     (2)
#endif
#ifndef CONFIG_CLOCK_PLL_MUL
#define CONFIG_CLOCK_PLL_MUL        (12)
#endif
```

I know the formula is not correct, just expressing an Idea. What do you think?

Reply:
Ok, but I prefer to add this comment in `f0/cfg_clock_default.h` before the `CLOCK_CORECLOCK` define.

Reply:
Otherwise, that means this has be added in all boards that need a custom value of the PREDIV. Maybe `CONFIG_CLOCK_PLL_PREDIV` should not be overridable on these platforms: they don't provide an HSE and PLLIN is always HSI / 2.

Reply:
Ok as you want.

Reply:
So I removed the overridable nature of `CONFIG_CLOCK_PLL_PREDIV` for f042 and f031 and added an explicit comment that explains the reasoning. That is much nicer this way. Thanks @fjmolinas for insisting ;)